### PR TITLE
Switching backend to firebase

### PR DESCRIPTION
**Pull request by merey90**
TODO: both players can move both teams, we have to set restrictions to avoid this behavior.

### CODE REVIEW

**Review by jarjan — APPROVED** [review]

File: public/index.html [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+    <title>ayaqdop</title>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="author" content="Olzhas Askar" />
+    <meta name="description" content="Chess-like football game." />
+    <link
+      rel="stylesheet"
+      href="https://fonts.googleapis.com/css?family=Overpass"
+    />
+    <base href="/" />
```
Comment (pheeria) [inline public/index.html:14]: just curious, what does this tag do?

Comment (merey90) [reply public/index.html:14]: it was impossible to import images in dynamic react-router-dom paths like `match\:matchId`

File: src/Ayaqdop/actions/movers.js [hunk 2]
```diff
@@ -1,44 +1,54 @@
...
+import { difference } from './helpers'
 
 export default function moveCore(gameObjects, piece, toPosition) {
+  let newMatchState = JSON.parse(JSON.stringify(gameObjects))
   if (piece.type === ItemTypes.BALL) {
-    moveBall(gameObjects, toPosition);
+    newMatchState = moveBall(newMatchState, toPosition)
   } else {
-    movePlayer(gameObjects, piece, toPosition);
+    newMatchState = movePlayer(newMatchState, piece, toPosition)
```
Comment (pheeria) [inline src/Ayaqdop/actions/movers.js:10]: nice pure functions! 👏

File: src/Ayaqdop/Field/Field.js [hunk 3]
```diff
@@ -1,55 +1,125 @@
...
+  const getSquareTarget = (column, row) => ({
+    canDrop(props, monitor) {
+      return canMoveCore(matchState, monitor.getItem(), [column, row])
+    },
+    drop(props, monitor) {
+      const newMatchState = moveCore(matchState, monitor.getItem(), [
+        column,
+        row
+      ])
+      emitChange(newMatchState)
```
Comment (pheeria) [inline src/Ayaqdop/Field/Field.js:65]: 🚀

**Review by pheeria — APPROVED** [review]

**Pull request merged**